ui/src/app/machines/views/AddMachine/AddMachineFormFields/AddMachineFormFields.js

```diff
@@ -124,6 +126,49 @@ export const AddMachineFormFields = () => {
           required
           type="text"
         />
+        {extraMACs.map((mac, i) => (
+          <div
+            className="p-input--closeable"
+            data-test={`extra-macs-${i}`}
+            key={`extra-macs-${i}`}
+          >
+            <Input
+              maxLength="17"
+              onChange={e => {
+                const newExtraMACs = [...extraMACs];
+                newExtraMACs[i] = e.target.value;
+                setExtraMACs(newExtraMACs);
+                setFieldValue("extra_macs", newExtraMACs);
+              }}
+              placeholder="00:00:00:00:00:00"
+              type="text"
+              value={mac}
+            />
+            <Button
+              className="p-close-input"
+              hasIcon
+              onClick={() => {
+                const newExtraMACs = extraMACs.filter((_, j) => j !== i);
+                setExtraMACs(newExtraMACs);
+                setFieldValue("extra_macs", newExtraMACs);
+              }}
+              type="button"
+            >
+              <i className="p-icon--close" />
+            </Button>
+          </div>
+        ))}
+        <div className="u-align--right">
+          <Button
+            data-test="add-extra-mac"
+            hasIcon
+            onClick={() => setExtraMACs([...extraMACs, ""])}
```

Comment:
Because new macs are set to an empty string these get included with the submitted data. It appears the API ignores them, but it might be nice to filter them out before the data is submitted.

Reply:
Good idea, adding a filter in now